converter/src/main/java/org/dswarm/converter/mf/stream/converter/JsonDecoder.java

```diff
@@ -0,0 +1,176 @@
+import org.culturegraph.mf.framework.annotations.Out;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.dswarm.converter.mf.framework.JsonReceiver;
+
+/**
+ * @author tgaengler
+ */
+@Description("Reads a JSON file and passes the JSON events to a receiver.")
+@In(Reader.class)
+@Out(JsonReceiver.class)
+public class JsonDecoder extends DefaultObjectPipe<Reader, JsonReceiver> {
+
+	private static final Logger LOG = LoggerFactory.getLogger(JsonDecoder.class);
+
+	private static final ObjectMapper MAPPER      = new ObjectMapper();
+	private static final JsonFactory  jsonFactory = MAPPER.getFactory();
+
+	private final Stack<String>    fieldNameStack;
+	private final Stack<JsonToken> tokenStack;
+
+	public JsonDecoder() {
+
+		super();
+
+		fieldNameStack = new Stack<>();
+		tokenStack = new Stack<>();
+	}
+
+	@Override
+	public void process(final Reader reader) {
+
+		try {
+
+			final JsonParser jp = jsonFactory.createParser(reader);
+
+			jp.nextToken();
+
+			while (jp.hasCurrentToken()) {
+
+				final JsonToken currentToken = jp.getCurrentToken();
+
+				switch (currentToken) {
+
+					case START_ARRAY:
+
+						tokenStack.push(currentToken);
+
+						break;
+					case START_OBJECT:
+
+						final String startObjectFieldName1 = getCurrentFieldName();
+
+						getReceiver().startObject(startObjectFieldName1);
+
+						tokenStack.push(currentToken);
+
+						break;
+					case END_ARRAY:
+
+						popFieldName();
+						tokenStack.pop();
+
+						if (JsonToken.FIELD_NAME.equals(tokenStack.peek())) {
+
+							tokenStack.pop();
+						}
+
+						break;
+					case END_OBJECT:
+
+						final String endObjectFieldName = getCurrentFieldName();
+						getReceiver().endObject(endObjectFieldName);
+
+						tokenStack.pop();
+
+						if (!tokenStack.isEmpty() && JsonToken.FIELD_NAME.equals(tokenStack.peek())) {
+
+							tokenStack.pop();
+						}
+
+						break;
+					case FIELD_NAME:
+
+						final String currentFieldName = jp.getCurrentName();
+
+						fieldNameStack.push(currentFieldName);
+						tokenStack.push(currentToken);
+
+						break;
+					case VALUE_FALSE:
+						//case VALUE_NULL:
+					case VALUE_NUMBER_FLOAT:
+					case VALUE_NUMBER_INT:
+					case VALUE_STRING:
+					case VALUE_TRUE:
+
+						final String currentValue = jp.getValueAsString();
+						final String fieldName = getCurrentFieldName();
+
+						getReceiver().literal(fieldName, currentValue);
+
+						if (JsonToken.FIELD_NAME.equals(tokenStack.peek())) {
+
+							popFieldName();
+							tokenStack.pop();
+						}
+
+						break;
+					case VALUE_NULL:
+
+						final String currentValue2 = jp.getValueAsString();
+						final String fieldName2 = getCurrentFieldName();
+
+						getReceiver().literal(fieldName2, currentValue2);
+
+						if (JsonToken.FIELD_NAME.equals(tokenStack.peek())) {
+
+							popFieldName();
+						}
+
+						tokenStack.pop();
+
+						break;
+					default:
+
+						// TODO: throw an exception (?)
```

Comment:
As long as Json isn't evolving and requiring Jackson to add new cases to the token enum, it doesn't really matter what to do in the default case, so just logging is fine to me.